File: MiniDOMBase/DMD/Anagrafica/Fonti/Canali/CCanaleCursor.cs
using System;
using DMD;
using DMD.Databases;
using static minidom.Sistema;
using minidom.repositories;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella dei canali
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CCanaleCursor 
            : minidom.Databases.DBObjectCursor<CCanale>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Tipo = new DBCursorStringField("Tipo");
            private DBCursorField<bool> m_Valid = new DBCursorField<bool>("Valid");
            private DBCursorStringField m_IconURL = new DBCursorStringField("IconURL");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCanaleCursor()
            {
            }

            /// <summary>
            /// Campo Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Campo Tipo
            /// </summary>
            public DBCursorStringField Tipo
            {
                get
                {
                    return m_Tipo;
                }
            }

            /// <summary>
            /// Campo IconURL
            /// </summary>
            public DBCursorStringField IconURL
            {
                get
                {
                    return m_IconURL;
                }
            }

            /// <summary>
            /// Campo Valid
            /// </summary>
            public DBCursorField<bool> Valid
            {
                get
                {
                    return m_Valid;
                }
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return APPConn;
            //}

            /// <summary>
            /// Restituisce un riferimento al repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.Canali; //.Module;
            }

            ///// <summary>
            ///// Restituisce il nome della tabella
            ///// </summary>
            ///// <returns></returns>
            //public override string GetTableName()
            //{
            //    return "tbl_ANACanali";
            //}
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Fonti/Canali/CCanaleCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
 
    ''' <summary>
    ''' Cursore sulla tabella dei canali
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CCanaleCursor
        Inherits DBObjectCursor(Of CCanale)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Tipo As New CCursorFieldObj(Of String)("Tipo")
        Private m_Valid As New CCursorField(Of Boolean)("Valid")
        Private m_IconURL As New CCursorFieldObj(Of String)("IconURL")

        Public Sub New()
        End Sub

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Tipo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Tipo
            End Get
        End Property

        Public ReadOnly Property IconURL As CCursorFieldObj(Of String)
            Get
                Return Me.m_IconURL
            End Get
        End Property

        Public ReadOnly Property Valid As CCursorField(Of Boolean)
            Get
                Return Me.m_Valid
            End Get
        End Property


        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Anagrafica.Canali.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_ANACanali"
        End Function

    End Class


End Class